prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchema.kt

```diff
@@ -58,45 +55,20 @@ data class ConfigSchema(
 
     /**
      * Validate the top level schema if [isChildSchema] is false, or a child schema if [isChildSchema] is true.
+     * [validationErrors] can optionally be specified to start with a list of errors, but defaults to an empty list.
      * @return a list of validation errors, or an empty list if no errors
      */
-    internal fun validate(isChildSchema: Boolean = false): List<String> {
-        val validationErrors = mutableListOf<String>()
-
+    internal open fun validate(
+        isChildSchema: Boolean = false,
+        validationErrors: MutableList<String> = mutableListOf()
```

Comment:
I imagine since it's not a private property, that it is best practice to keep it immutable. But you're right that it would simplify things, maybe we could have a private property that is a mutable list which can be used by the validation functions, and the non-private immutable list can stay as a separate property?

Reply:
@carlosfelix2 the reason for making it an argument was to allow the child classes to do their domain specific validation, and then still call `super.validate` and pass through the list of errors so far.

Reply:
Adjusted this in latest commit